File: python/73. Set Matrix Zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        row = [0] * len(matrix)
        col = [0] * len(matrix[0])
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    row[i] = col[j] = 1
        for i in range(len(row)):
            if row[i] == 1:
                for j in range(len(col)):
                    matrix[i][j] = 0
        for i in range(len(col)):
            if col[i] == 1:
                for j in range(len(row)):
                    matrix[j][i] = 0
        

File: cpp/51-100/73. Set Matrix Zeroes.cpp
class Solution {
public:
	void setZeroes(vector<vector<int>>& matrix)
	{
		bool row = false, col = false;
		for (int j = 1; j < matrix[0].size(); j++)
			if (matrix[0][j] == 0)
				row = true;
		for (int i = 1; i < matrix.size(); i++)
			if (matrix[i][0] == 0)
				col = true;
		for (int i = 1; i < matrix.size(); i++) {
			for (int j = 1; j < matrix[0].size(); j++)
				if (matrix[i][j] == 0) {
					matrix[0][j] = matrix[i][0] = 0;
				}
		}
		for (int i = 1; i < matrix.size(); i++) {
			if (matrix[i][0] == 0) {
				vector<int> temp(matrix[0].size(), 0);
				matrix[i] = temp;
			}
		}
		for (int j = 1; j < matrix[0].size(); j++) {
			if (matrix[0][j] == 0)
			{
				for (int i = 1; i < matrix.size(); i++)
					matrix[i][j] = 0;
			}
		}
		if (matrix[0][0] == 0)
		{
			vector<int> temp(matrix[0].size(), 0);
			matrix[0] = temp;
			for (int i = 0; i < matrix.size(); i++)
				matrix[i][0] = 0;
		}
		if (row) {
			vector<int> temp(matrix[0].size(), 0);
			matrix[0] = temp;
		}
		if (col)
		{
			for (int i = 0; i < matrix.size(); i++)
				matrix[i][0] = 0;
		}
	}
};